test/option-nonce.test.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const jwt = require('../');
+const expect = require('chai').expect;
+const util = require('util');
+
+describe('nonce option', function () {
+  let token;
+
+  beforeEach(function () {
+    token = jwt.sign({ nonce: 'abcde' }, undefined, { algorithm: 'none' });
+  });
+  [
+    {
+      description: 'should work with string',
+      nonce: 'abcde',
+    },
+  ].forEach((testCase) => {
+    it(testCase.description, function (done) {
+      expect(jwt.verify(token, undefined, { nonce: testCase.nonce })).to.not.throw;
+      jwt.verify(token, undefined, { nonce: testCase.nonce }, (err) => {
+        expect(err).to.be.null;
+        done();
+      })
+    });
```

Comment:
Can you do the tests using the helpers that @MitMaro prepared?
It will internally assert sync and async behavior (like you did well here, just for consistency across tests)

Reply:
@kazuki229 Some direction that hopefully will make things a little easier.

Add `const testUtils = require('./test-utils');` to get the test utilities.

To use the test helpers for `verify`:

```
testUtils.verifyJWTHelper(token, undefined, {nonce: testCase.nonce}, (err, decoded) => {
  testUtils.asyncCheck(done, () => {
    expect(err).to.be.null;
    expect(decoded).to.have.property('nonce', 'abcde');
  });
});
```

To check for an error you would do something along the lines of:

```
testUtils.verifyJWTHelper(token, undefined, {nonce}, (err) => {
  testUtils.asyncCheck(done, () => {
    expect(err).to.be.instanceOf(jwt.JsonWebTokenError);
    expect(err).to.have.property('message', 'nonce must be a non-empty string')
  });
});
```

Ping me if you need any other assistance and I will gladly help!